(Docs) getStaticProps & getServerSideProps TS section (#40607) (#40639)
vercel/next.js: docs/api-reference/data-fetching/get-static-props.md

## getStaticProps with TypeScript

> To specify type of `getStaticProps` explicitly, you can use `GetStaticProps` from `next`:

Reviewer: ```suggestion
The type of `getStaticProps` can be specified using `GetStaticProps` from `next`:
```

```ts
import { GetStaticProps } from 'next'

type Post = {
  author: string
  content: string
}

export const getStaticProps: GetStaticProps<{ posts: Post[] }> = async (
  context
) => {
  const res = await fetch('https://.../posts')
  const posts: Post[] = await res.json()

  return {
    props: {
      posts,
    },
  }
}
```

If you want to get inferred typings for your props, you can use `InferGetStaticPropsType<typeof getStaticProps>`:

> ```tsx

Reviewer 2: Will tsx work? I am unsure (worth checking)
